src/sentry_core.c

```diff
@@ -545,6 +561,15 @@ sentry__prepare_event(const sentry_options_t *options, sentry_value_t event,
         if (!item) {
             continue;
         }
+        if (attachment->type) {
```

Comment:
While this is technically correct, since the backend seemingly auto-assigns `event.attachment` to envelopes without an `attachment_type`, I am not sure that relying on the numeric representation is the right approach.

We actually have to check `content_type` because it could be `NULL`, which has a meaning (we want auto-detection in the backend).

For `attachment_type`, we rely on the value being there so we can add the header unconditionally. If you want to express that we don't need to explicitly send the `attachment_type` if it is `event.attachment` because it is the default, then let's compare explicitly:

```suggestion
        if (attachment->type != ATTACHMENT) {
```